templates/statefulset.yaml

```diff
@@ -0,0 +1,159 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "keycloak.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    {{- range $key, $val := .Values.statefulSet.annotations }}
+    {{- printf "%s: %s" $key ($val | quote) | nindent 4 }}
+    {{- end }}
+  labels:
+    {{- range $key, $val := .Values.statefulSet.labels }}
+    {{- printf "%s: %s" $key ($val | quote) | nindent 4 }}
+    {{- end }}
+    {{- include "keycloak.labels" . | nindent 4 }}
+spec:
+  serviceName: 
+  selector:
+    matchLabels:
+      {{- include "keycloak.selectorLabels" . | nindent 6 }}
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  podManagementPolicy: {{ .Values.podManagementPolicy }}
+  updateStrategy:
+    type: {{ .Values.statefulSet.updateStrategy }}
+  template:
+    metadata:
+      annotations:
+        {{- range $key, $val := .Values.podAnnotations }}
+        {{- printf "%s: %s" $key ($val | quote) | nindent 8 }}
+        {{- end }}
+      labels:
+        {{- range $key, $val := .Values.podLabels }}
+        {{- printf "%s: %s" $key ($val | quote) | nindent 8 }}
+        {{- end }}
+        {{- include "keycloak.selectorLabels" . | nindent 8 }}
+    spec:
+      containers:
+      - name: keycloak
+        {{- if .Values.image.digest}}
+        image: "{{ .Values.image.repository }}@{{ .Values.image.digest }}"
```

Comment:
isn't the tag missing here if you want to pull e.g. `quay.io/keycloak/keycloak:22.0.1@sha256:9870...`

see https://stackoverflow.com/a/63204994/9710727

Reply:
Apparently, if a digest is specified, the tag is ignored. `docker image pull quay.io/keycloak/keycloak:notexistingtag@sha256:5b872e8...` works regardless of which tag is used.